### ISSUE TITLE
Smarter default values for tool settings (trace width, via size, ...)

### ISSUE THREAD

**Issue by ubruhin** [LibrePCB/LibrePCB#1354]
##### SUMMARY

Currently the default settings of tools (e.g. trace width or via size) are hardcoded and might often not be what is desired by the user. See also https://librepcb.discourse.group/t/board-editor-keeps-forgetting-my-settings/719.

##### SOLUTION

Instead of hardcoded values, we might consider:
- Providing values already used in the opened project
- Memorize and restore last used values after application restart
- Make default values configurable

**Comment by ubruhin** [comment]
#1610 and #1611 highly improve that situation. I think it could be improved further by memorizing the trace/via settings (from the toolbar) in the board user settings file, but that's now low priority.

**Comment by SiboVG** [comment]
I'm not sure if it fits this issue, but I was quite surprised that the draw trace tool defaults to 90° angles. My standard, and that of many other PCB designers (I think), is to use 45° angles. It helps minimize board space, and is better for high-frequency signal integrity. 90° angles also used to be a danger for acid traps during manufacturing, but I'm not sure if that's the case these days.

In any case, I propose to default to 45° angles.

**Comment by ubruhin** [comment]
> I'm not sure if it fits this issue, but I was quite surprised that the draw trace tool defaults to 90° angles.

Good point, I have never been thinking about this. I suspect the default mode was just copied from the schematic editor (where 90° is a good default IMO) and didn't realize it's not a good default for the board.

> In any case, I propose to default to 45° angles.

:+1:
